lib/onelogin/ruby-saml/response.rb

```diff
@@ -51,6 +53,21 @@ def initialize(response, options = {})
 
         @response = decode_raw_saml(response)
         @document = XMLSecurity::SignedDocument.new(@response, @errors)
+
+        return unless assertion_encrypted?
```

Comment:
@luisvm could be that a solution?
https://gist.github.com/pitbulk/c143ddc83aeaefca3834